src/core/opamFilename.ml

```diff
@@ -9,6 +9,10 @@
 (*                                                                        *)
 (**************************************************************************)
 
+let might_escape path =
+  List.exists (String.equal Filename.parent_dir_name)
+    Re.(split (compile (str Filename.dir_sep)) path)
```

Comment:
I always struggle to understand `Re` (and regular expressions in general) -- I understand that Filename.dir_sep is returning a string, and String.split_on_char expects a character to split on.

But since all known values of Filename.dir_sep are a single-character string, I'd go down that path.

(of course, this is just a minor thing, and please ignore if you're happy with regular expressions).